feat(ProjectUI): Sort & Filter for ClearingStatus TreeView table

akapti opened the pull request:
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
* Added the `sort` and `search` functionality for `Clearing Status` _Tree View_ table.

> Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
* #1102 

>  Did you add or update any new dependencies that are required for your change?
* No

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

### How To Test?
> How should these changes be tested by the reviewer?
* Open the project with `linked releases` and `linked projects`, and go to `clearing status` tab `TreeView` table.
* Sort the table based on rows by clicking on header. (except `state`, `comment` and `action` row)
* Search full table (except `action` row)
* Filter table based on `Release Clearing Status`.

Screenshot for reference:
![image](https://user-images.githubusercontent.com/50870174/109300018-fe601980-785b-11eb-8d27-33f615b52efc.png)


> Have you implemented any additional tests?
* No

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>

JaideepPalit commented on frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/clearingStatus.jsp:
```diff
@@ -45,6 +45,9 @@
 <div class="tab-content" id="pills-clearingStatusTab">
     <div class="tab-pane fade show active" id="pills-treeView" role="tabpanel" aria-labelledby="pills-tree-tab">
+    <div class="float-right mx-2">
+        <input type="text" id="search_table" placeholder="Search...">
```
Since `Search...` is a hardcoded values.
Kindly use language.properties for I18N

Would it be possible to use bootstrap class `form-control` like it is used in other places like clearing request etc.
So that textbox looks similar in entire sw360

akapti replied:
Used `language.properties` for Search placeholder, and `form-control` class for Search input.

JaideepPalit reviewed (changes requested):
Kindly look into some minor review comments.
Tested the PR. Seems to be some issues.
- Issue 1:
Sort indicator dos not indicate whether it is sort by ascending or descending with single caret.
![image](https://user-images.githubusercontent.com/56534710/108491467-f5a29d00-72c9-11eb-95b8-47681d45809f.png)
- Issue 2:
Sort does not work perfectly like while sorting by name, `CLI` comes in middle of of `ExcelComp2` and `ProjectTodo4`, similar thing observed for `Project Mainline State`
![image](https://user-images.githubusercontent.com/56534710/108491233-a8263000-72c9-11eb-9ec0-dff455d1549f.png)

![image](https://user-images.githubusercontent.com/56534710/108491580-18cd4c80-72ca-11eb-8b62-85800a9e9389.png)

- Issue 3
No message when nothing fulfils the search criteria
![image](https://user-images.githubusercontent.com/56534710/108491393-d99efb80-72c9-11eb-80b4-4be249bc81c9.png)
- Issue 4
Sort functionality work for level 1 nodes of tree view and not for child nodes.
![image](https://user-images.githubusercontent.com/56534710/108491958-95f8c180-72ca-11eb-90ba-a04a4145a4ec.png)
- Issue 5
Search functionality does not work properly for child nodes.
Original
![image](https://user-images.githubusercontent.com/56534710/108492480-39e26d00-72cb-11eb-8650-12ff6a78d6e9.png)
After Search, parent node disappears
![image](https://user-images.githubusercontent.com/56534710/108492636-66968480-72cb-11eb-93e6-1c94cc877e54.png)

akapti commented:
> * Issue 1:
>   Sort indicator dos not indicate whether it is sort by ascending or descending with single caret.
* Added `Ascending` / `descending` sort indicator.

> * Issue 2:
>   Sort does not work perfectly like while sorting by name, `CLI` comes in middle of of `ExcelComp2` and `ProjectTodo4`, similar thing observed for `Project Mainline State`
* Fixed the issue.

> * Issue 3
>   No message when nothing fulfils the search criteria
* Added the message `No linked releases or projects` when search results are empty.

> * Issue 4
>   Sort functionality work for level 1 nodes of tree view and not for child nodes.
* This is expected behavior, child nodes are always sorted in ascending order, as it gets complex to apply `search` and `sort` at same time for `treetable`.

> * Issue 5
>   Search functionality does not work properly for child nodes.
* Fixed, searching for the `child` rows, will display complete hierarchy of it's parents.

akapti commented:
added the feature to filter based on `Release Clearing State`.

JaideepPalit reviewed (approved):
Code looks fine to me. Tested successfully.
